lib/widgets/read_receipts.dart

```diff
@@ -0,0 +1,208 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../api/route/messages.dart';
+import '../generated/l10n/zulip_localizations.dart';
+import 'action_sheet.dart';
+import 'actions.dart';
+import 'color.dart';
+import 'inset_shadow.dart';
+import 'profile.dart';
+import 'store.dart';
+import 'text.dart';
+import 'theme.dart';
+import 'user.dart';
+
+/// Opens a bottom sheet showing who has read the message.
+void showReadReceiptsSheet(BuildContext pageContext, {required int messageId}) {
+  final accountId = PerAccountStoreWidget.accountIdOf(pageContext);
+
+  showModalBottomSheet<void>(
+    context: pageContext,
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
+    builder: (_) {
+      return PerAccountStoreWidget(
+        accountId: accountId,
+        child: SafeArea(
+          minimum: const EdgeInsets.only(bottom: 16),
+          child: ReadReceipts(messageId: messageId)));
+    });
+}
+
+/// The read receipts sheet.
+///
+/// Figma link:
+///   https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=11367-20647&t=lSnHudU6l7NWx0Fa-0
+class ReadReceipts extends StatefulWidget {
+  const ReadReceipts({super.key, required this.messageId});
+
+  final int messageId;
+
+  static const _helpCenterReference = '/help/read-receipts';
+
+  @override
+  State<ReadReceipts> createState() => _ReadReceiptsState();
+}
+
+class _ReadReceiptsState extends State<ReadReceipts> with PerAccountStoreAwareStateMixin<ReadReceipts> {
+  List<int> userIds = [];
+  FetchStatus status = FetchStatus.loading;
+
+  @override
+  void onNewStore() {
+    tryFetchReadReceipts(context);
+  }
+
+  Future<void> tryFetchReadReceipts(BuildContext context) async {
+    final store = PerAccountStoreWidget.of(context);
+    try {
+      final result = await getReadReceipts(store.connection, messageId: widget.messageId);
+
+      if (!context.mounted) return;
+      final storeNow = PerAccountStoreWidget.of(context);
+      if (!identical(store, storeNow)) return;
+
+      // TODO(i18n): add locale-aware sorting
+      userIds = result.userIds.sortedByCompare(
+        (id) => storeNow.userDisplayName(id),
```

Comment:
Since `store` and `storeNow` are identical, this is equivalent:
```suggestion
        store.userDisplayName,
```

and keeps to our convention of always calling the store just `store`.

(This version also incorporates a nit that this can use a tearoff instead of a fresh closure.)